src/term/transform/flatten.rs

```diff
@@ -188,11 +188,7 @@ fn make_split_rule(old_rule: &Rule, other_rule: &Rule, def_names: &DefNames) ->
         let new_ctr = Term::call(Term::Ref { def_id: rule_arg_def_id }, new_ctr_args);
         new_body.subst(other_arg, &new_ctr);
       }
-      (Pattern::Num(..), Pattern::Num(..)) => new_pats.push(other_arg.clone()),
-      (Pattern::Num(..), Pattern::Var(..)) => {
-        // How to do this with this kind of number pattern? Subst with a match?
-        todo!();
-      }
+      (Pattern::Num(..), _) => unreachable!(),
```

Comment:
Can you add a comment explaining that this is unreacheable because `Num` can't have nested patterns?